DribbbleSDK/Client/DRApiClient.m

```diff
@@ -146,21 +145,30 @@ - (AFHTTPRequestOperation *)createRequestWithMethod:(NSString *)method requestTy
             responseHandler([weakSelf mappedDataFromResponseObject:responseObject modelClass:modelClass]);
         }
     } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+        if ([operation.response statusCode] == 404) {
+            NSError *dribbbleFalseError = [NSError errorWithDomain:error.domain code:404 userInfo:error.userInfo];
+            error = dribbbleFalseError;
+        }
         if (weakSelf.defaultErrorHandler) weakSelf.defaultErrorHandler(error);
         if (responseHandler) responseHandler([DRApiResponse responseWithError:error]);
     }];
     return operation;
 }
 
-- (void)runMultiPartRequestWithMethod:(NSString *)method params:(NSDictionary *)params data:(NSData *)data mimeType:(NSString *)mimeType responseHandler:(DRResponseHandler)responseHandler {
-    UIImage *image = [[UIImage alloc] initWithData:data];
-    CGSize imageSize = image.size;
-    NSAssert((imageSize.width == 400.f && imageSize.height == 300.f) || (imageSize.width == 800.f && imageSize.height == 600.f), kUploadImageSizeAssertionString);
-    NSAssert((data.length/1024.f/1024.f) <= kUploadFileBytesLimitSize, kUploadFileSizeAssertionString);
-    
+- (void)runMultiPartRequestWithMethod:(NSString *)method params:(NSDictionary *)params data:(NSData *)data mimeType:(NSString *)mimeType isAttachment:(BOOL)isAttachment responseHandler:(DRResponseHandler)responseHandler {
```

Comment:
refactor. add parameter "parameterName" in this method, remove isAttachment condition (the main point is: keep generic method generic). add two methods like: uploadAttachment and uploadFile, and these methods should call generic method with differrent value of parameterName